pandas/_libs/tslibs/parsing.pyx

```diff
@@ -976,7 +976,6 @@ def guess_datetime_format(dt_str: str, bint dayfirst=False) -> str | None:
         (('hour',), '%H', 2),
         (('minute',), '%M', 2),
         (('second',), '%S', 2),
-        (('microsecond',), '%f', 6),
```

Comment:
this doesn't seem to make a difference - at least, I don't think it's covered by existing tests